### PR TITLE
indexer: manifest deletion

### PR DESCRIPTION
**Pull request by hdonnay**
These changes modify the database to add `CASCADE` constraints on some foreign keys and exposes + implements API surface to remove manifests from the underlying indexer data store. This is needed to prevent the indexer's database from growing unbounded.

Along the way there were some refactors, clean ups, and helpers.

### CODE REVIEW

File: libindex/migrations/04-foreign-key-cascades.sql [hunk 1]
```diff
@@ -0,0 +1,165 @@
+-- Add CASADEs on all foreign-key relations.
```
Comment (crozzy) [inline libindex/migrations/04-foreign-key-cascades.sql:1]: Do we want to lock the tables? I'm trying to think of someone is cycling a bunch of indexer pods with new versions while the old ones are still working.

File: libindex/migrations/migrations.go [hunk 2]
```diff
@@ -2,34 +2,44 @@ package migrations
...
 	"github.com/remind101/migrate"
 )
 
-const (
-	MigrationTable = "libindex_migrations"
-)
+//go:embed *.sql
+var fs embed.FS
+
+func runFile(n string) func(*sql.Tx) error {
```
Comment (crozzy) [inline libindex/migrations/migrations.go:13]: Much nicer

**Review by crozzy — COMMENTED** [review]
A couple of comments, but overall LGTM

File: libindex/migrations/04-foreign-key-cascades.sql (hunk 1 above)
Comment (hdonnay) [reply libindex/migrations/04-foreign-key-cascades.sql:1]: according to the docs, the PostgresQL migrator takes an advisory lock

**Review by crozzy — APPROVED** [review]

**Pull request merged**